lib/auth/auth_login_test.go

```diff
@@ -773,61 +773,192 @@ func TestCreateRegisterChallenge_unusableDevice(t *testing.T) {
+		_, err := s.AuthenticateWebUser(ctx, req)
+		return err
+	}))
+}
+
+func TestServer_AuthenticateUser_passwordOnly_failure(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	svr := newTestTLSServer(t)
+	authServer := svr.Auth()
+
+	setPassword := func(pwd string) func(*testing.T, *Server) {
+		return func(t *testing.T, s *Server) {
+			require.NoError(t, s.UpsertPassword("capybara", []byte(pwd)))
+		}
+	}
 
 	tests := []struct {
-		name                 string
-		solveChallenge       func(*configureMFAResp, *proto.MFAAuthenticateChallenge) (*proto.MFAAuthenticateResponse, error)
-		initialPasswordState *types.PasswordState
+		name         string
+		setup        func(*testing.T, *Server)
+		authUser     string
+		authPassword string
 	}{
 		{
-			name: "OK TOTP device",
-			solveChallenge: func(mfa *configureMFAResp, c *proto.MFAAuthenticateChallenge) (*proto.MFAAuthenticateResponse, error) {
-				return mfa.TOTPDev.SolveAuthn(c)
-			},
+			name:         "wrong password",
+			setup:        setPassword("secure password"),
+			authUser:     "capybara",
+			authPassword: "wrong password",
 		},
 		{
-			name: "OK Webauthn device",
-			solveChallenge: func(mfa *configureMFAResp, c *proto.MFAAuthenticateChallenge) (*proto.MFAAuthenticateResponse, error) {
-				return mfa.WebDev.SolveAuthn(c)
-			},
+			name:         "user not found",
+			setup:        setPassword("secure password"),
+			authUser:     "unknown",
+			authPassword: "secure password",
 		},
 		{
-			name: "OK TOTP device (legacy)",
-			solveChallenge: func(mfa *configureMFAResp, c *proto.MFAAuthenticateChallenge) (*proto.MFAAuthenticateResponse, error) {
-				return mfa.TOTPDev.SolveAuthn(c)
-			},
-			initialPasswordState: &psUnspecified,
+			name:         "password not found",
+			setup:        func(*testing.T, *Server) {},
+			authUser:     "capybara",
+			authPassword: "secure password",
 		},
 	}
-	for _, test := range tests {
-		test := test
 
+	for _, test := range tests {
 		// makeRun is used to test both SSH and Web login by switching the
 		// authenticate function.
 		makeRun := func(authenticate func(*Server, AuthenticateUserRequest) error) func(t *testing.T) {
 			return func(t *testing.T) {
-				svr := newTestTLSServer(t)
-				authServer := svr.Auth()
-				mfa := configureForMFA(t, svr)
-				username := mfa.User
-				password := mfa.Password
-
-				if test.initialPasswordState != nil {
-					// Enforce a specific password state.
-					u, err := authServer.Identity.UpdateAndSwapUser(ctx, username, false, /* withSecrets */
-						func(u types.User) (bool, error) {
-							u.SetPasswordState(*test.initialPasswordState)
-							return true, nil
-						})
-					require.NoError(t, err)
-					assert.Equal(t, *test.initialPasswordState, u.GetPasswordState())
-				}
+				_, _, err := CreateUserAndRole(authServer, "capybara", nil, nil)
+				require.NoError(t, err)
+				defer authServer.DeleteUser(ctx, "capybara")
```

Comment:
Suggestion: Use a cleanup, handle error.

```suggestion
				t.Cleanup(func() {
					assert.NoError(t,
						authServer.DeleteUser(ctx, "capybara"),
						"Failed to delete the capybara user")
        })
```